modules/custom/openy_repeat/src/RepeatManager.php

```diff
@@ -280,14 +280,20 @@ public static function calcSessionInstancesBySchedule(array $session_schedule, $
         $exclusions = self::reorderExclusions($session_schedule['exclusions']);
         $combined_dates = self::combineDates($dates, $exclusions);
 
+        $user_time_zone = drupal_get_user_timezone();
```

Comment:
@anpolimus `drupal_get_user_timezone()` doesn't seem to be a right choice here.
See the [documentation](https://api.drupal.org/api/drupal/core%21includes%21bootstrap.inc/function/drupal_get_user_timezone/8.2.x)

The function returns current user timezone instead of the website timezone. So if a logged-in user with overridden timezone setting updates the schedule somehow (e.g. renames a class or a program or a category), the schedule will be rebuilt with the wrong timezone in mind.

Reply:
You are totally right. Thank you.

Reply:
@podarok @AndreyMaximov 
I wanted to make this code changes implemented according to best Drupal coding standards but faced with a problem that I cant use $this->configFactory in this case (public static function calcSessionInstancesBySchedule - static method).
I dont want to break existed functionality and leave using of Drupal:: config instead.

Reply:
@AndreyMaximov code updated.

Reply:
@anpolimus Correct, in static methods you should call services using `\Drupal::service()` (or shortcuts like `\Drupal::config()`). That's totally fine.

Thanks.

Reply:
calling those methods at static calls are making static method not static in real life :)